packages/components/switch/src/Switch.tsx

```diff
@@ -6,12 +8,45 @@ import { Label } from './SwitchLabel'
 export type SwitchProps = InputProps
 
 export const Switch = forwardRef<HTMLButtonElement, SwitchProps>(
-  ({ value = 'on', size = 'md', children, className, ...rest }, ref) => {
+  ({ value = 'on', size = 'md', children, className, id, disabled, ...rest }, ref) => {
+    const { id: controlledInputId, labelId: controlledLabelId } = useFormFieldState()
+
+    const switchId = useId(id)
+    const switchLabelId = useId()
+
     return (
-      <Label data-spark-component="switch" disabled={rest.disabled} className={className}>
-        <Input ref={ref} size={size} {...rest} />
-        <div className="block">{children}</div>
-      </Label>
+      <div data-spark-component="switch" className="gap-md text-body-1 flex items-center">
+        <Input
+          ref={ref}
+          size={size}
+          id={controlledInputId || switchId}
+          disabled={disabled}
+          /**
+           * If the switch doesn't have any direct label (children) then we should try to
+           * get an eventual alternative label from FormField.
+           * On last resort, we shouldn't forget to define an aria-label attribute.
+           */
```

Comment:
IMO tests should explain this part

Reply:
Wdyt? https://github.com/adevinta/spark/commit/576488518abba5e747a31386ac92ce9a97d3da0a